graphql/data/schema.js

```diff
@@ -965,6 +968,87 @@ const CauseOnboardingCopyType = new GraphQLObjectType({
     },
   }),
 })
+
+const impactMetricType = new GraphQLObjectType({
+  name: 'ImpactMetric',
+  description: 'An instance of GroupImpact',
+  fields: () => ({
+    id: {
+      type: globalIdField(IMPACT_METRIC),
+      description: 'ID of the ImpactMetric',
+    },
+    charityId: {
```

Comment:
Use Charity type directly here. (Rule of thumb, if referencing the ID of another object, just use the object, which allows the client to query on additional fields from that object.)

Reply:
I initially didn't want to do that as these entities would be returned as a field of the Charity type. Wouldn't that potentially cause some sort of circular dependency?

Reply:
I don't see how it would cause a problem. The resolver won't be called unless the field is queried, so I think the only way it would be kind-of circular is if someone made an intentionally deep nested query. If someone did query `charity > impact metric > charity` (I'd hope we'd catch this in code review), it should just resolve.

On the point of "would be returned as a field of the Charity type" -- this is true for what we can predict now, but I don't think the types need to make assumptions about their parent entities.

LMK if you feel I'm missing something!

Reply:
Ah, thank you for the explanation! That totally makes sense. Thanks.